### PR TITLE
style: rename params for clearity, update && added comments

### PR DESCRIPTION
**Pull request by gandazgul**
### `Noteable Changes`

- Updated docs
- Added some comments
- Renamed params to make them clearer

### CODE REVIEW

File: src/MakeLocalizeFunction.js [hunk 1]
```diff
@@ -10,25 +13,33 @@ function makeLocalizeFunction(localization, nested) {
...
+ *
+ * @param {object} localization
+ * @param {string} nestedKey The original key
  *
- * @param {object}  localization
- * @param {string}  string key
  * @returns {*}
  */
-function byString(object, string) {
+function byString(localization, nestedKey) {
```
Comment (michael-ciniawsky) [inline src/MakeLocalizeFunction.js]: `nestedkey` => `key`

File: src/MakeLocalizeFunction.js [hunk 2]
```diff
@@ -10,25 +13,33 @@ function makeLocalizeFunction(localization, nested) {
...
   // strip a leading dot
-  const stringKey = string.replace(/^\./, '');
-
+  const stringKey = nestedKey.replace(/^\./, '');
```
Comment (michael-ciniawsky) [inline src/MakeLocalizeFunction.js]: `key = key.replace(/^\./, '');`

File: src/MakeLocalizeFunction.js [hunk 3]
```diff
@@ -10,25 +13,33 @@ function makeLocalizeFunction(localization, nested) {
...
   const keysArray = stringKey.split('.');
```
Comment (michael-ciniawsky) [inline src/MakeLocalizeFunction.js]: `keysArray` => `keys`

File: src/MakeLocalizeFunction.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
 /**
+ * Convert the localization object into a function in case we need to support nested keys.
  *
- * @param {object}  localization
- * @returns {Function}
+ * @param {object} localization the language object,
```
Comment (michael-ciniawsky) [inline src/MakeLocalizeFunction.js]: `{type}` => `{Type}`

Comment (gandazgul) [reply src/MakeLocalizeFunction.js]: I thought about it, but saw lowercase everywhere so left it. I can change this.

File: src/MakeLocalizeFunction.js (hunk 1 above)
Comment (gandazgul) [reply src/MakeLocalizeFunction.js]: key is a variable used in the function, I rather not param reassign if I dont have to. nestedKey makes sense from what the function does.

File: src/MakeLocalizeFunction.js (hunk 2 above)
Comment (gandazgul) [reply src/MakeLocalizeFunction.js]: see my other comment.

File: src/MakeLocalizeFunction.js (hunk 3 above)
Comment (gandazgul) [reply src/MakeLocalizeFunction.js]: sure I'll rename this.

File: src/MakeLocalizeFunction.js (hunk 4 above)
Comment (michael-ciniawsky) [reply src/MakeLocalizeFunction.js]: Yep would be appreciated while we are at it 😛

Comment (gandazgul) [reply src/MakeLocalizeFunction.js]: done

File: src/MakeLocalizeFunction.js (hunk 3 above)
Comment (gandazgul) [reply src/MakeLocalizeFunction.js]: done

File: src/MakeLocalizeFunction.js [hunk 5]
```diff
@@ -10,25 +13,32 @@ function makeLocalizeFunction(localization, nested) {
...
  *
- * @param {object}  localization
- * @param {string}  string key
  * @returns {*}
  */
-function byString(object, string) {
-  // strip a leading dot
-  const stringKey = string.replace(/^\./, '');
+function byString(localization, nestedKey) {
+  // remove a leading dot and split by dot
```
Comment (michael-ciniawsky) [inline src/MakeLocalizeFunction.js]: `// remove the leading dot and split ...`

**Comment by gandazgul** [comment]
Can we merge this in?

**Pull request merged**